api/grpc/controller_test.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package grpc
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	tmock "github.com/stretchr/testify/mock"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/models/dps/mock"
+)
+
+func TestNewController(t *testing.T) {
+	c := NewController(nil)
+	assert.NotNil(t, c)
```

Comment:
This is an example of where I'm in favour of having unit tests in the same package. The constructor test should always test if all fields are successfully set and the configuration is successfully injected.